### PR TITLE
fix warnings found on 16.04

### PR DESCRIPTION
**Pull request by hendrikmuhs**
@narekgharibyan 

some small type fixes to get rid of all warnings on gcc-5.4 (ubuntu 16.04)

### CODE REVIEW

File: keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h [hunk 1]
```diff
@@ -265,7 +265,7 @@ final {
             }
 
             // check if l has final info
-            int value = persistence_->ReadFinalValue(l.GetOffset());
+            uint64_t value = persistence_->ReadFinalValue(l.GetOffset());
```
Comment (narekgharibyan) [inline keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h]: As value is only used for reading we can declare it as `const`

File: keyvi/src/cpp/dictionary/fsa/internal/bit_vector_64.h [hunk 2]
```diff
@@ -84,12 +84,12 @@ final {
 
       if (bitPosition == 0)
       {
-        for (auto i = 0; i<write_length; i++){
+        for (size_t i = 0; i<write_length; i++){
```
Comment (narekgharibyan) [inline keyvi/src/cpp/dictionary/fsa/internal/bit_vector_64.h]: In general it's a good practice to do pre-increment (++i) if we don't need the old value, as we avoid a temporary variable creation.

File: keyvi/src/cpp/dictionary/fsa/internal/bit_vector_64.h [hunk 3]
```diff
@@ -84,12 +84,12 @@ final {
...
           bits_[bytePosition + i] |= other.bits_[i];
         }
       } else {
         bits_[bytePosition] |= ((uint64_t) other.bits_[0]<< bitPosition);
-        for (auto i = 1; i<write_length; i++){
+        for (size_t i = 1; i<write_length; i++){
```
Comment (narekgharibyan) [inline keyvi/src/cpp/dictionary/fsa/internal/bit_vector_64.h]: use pre-increment

File: keyvi/src/cpp/dictionary/fsa/internal/bit_vector_64.h [hunk 4]
```diff
@@ -108,7 +108,7 @@ final {
 
       size_t write_length = std::min(bits_.size(), other.bits_.size()) - bytePosition_other;
 
-      for (auto i = 0; i<write_length; i++){
+      for (size_t i = 0; i<write_length; i++){
```
Comment (narekgharibyan) [inline keyvi/src/cpp/dictionary/fsa/internal/bit_vector_64.h]: use pre-increment

File: keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h (hunk 1 above)
Comment (hendrikmuhs) [inline keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h]: which is not the case

Comment (narekgharibyan) [inline keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h]: @hendrikmuhs I didn't get you comment.

Comment (hendrikmuhs) [inline keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h]: I thought you mean some lines above where we re-use value

Comment (narekgharibyan) [inline keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h]: Ah, no. I meant variable `uint64_t value`

Comment (hendrikmuhs) [inline keyvi/src/cpp/dictionary/fsa/internal/unpacked_state.h]: ok

**Comment by hendrikmuhs** [comment]
did the changes from review

(BTW. pre-increment vs. post-increment only matters for none-simple types, changed it anyway as it is good pattern)

**Pull request merged**